synapse/federation/federation_server.py

```diff
@@ -645,19 +646,40 @@ async def on_invite_request(
         return {"event": ret_pdu.get_pdu_json(time_now)}
 
     async def on_send_join_request(
-        self, origin: str, content: JsonDict, room_id: str
+        self,
+        origin: str,
+        content: JsonDict,
+        room_id: str,
+        caller_supports_partial_state: bool = False,
     ) -> Dict[str, Any]:
         event, context = await self._on_send_membership_event(
             origin, content, Membership.JOIN, room_id
         )
 
         prev_state_ids = await context.get_prev_state_ids()
-        state_event_ids = prev_state_ids.values()
+
+        state_event_ids: Collection[str]
+        servers_in_room: Optional[Collection[str]]
```

Comment:
> I guess also because the joining homeserver might need to send events and for that, it needs to know which servers are in the room 

yeah, this is the main driver right now.